### ISSUE TITLE
Events keep re-executing indefinitely, throttling EventHandler for other events

### ISSUE THREAD

**Issue by danielHalan** [Particular/NServiceBus.AzureServiceBus#712]
Hi, 

Since yesterday there has been a lot of timeout messages at a Event Handler, where the root cause is unknown if its on MS ASB side, as there is no code changes and the events it occurs on have been somewhat random. 

But what happens afterwards is that the failing event(s) is being continuously re-executed, in parallel and preventing other events being processed. The Max Delivery Count in ASB has been set to a high number to expect the NSB error-handling to take over, which has following settings MaxRetries=4 MaximumConcurrencyLevel=12 MaximumMessageThroughputPerSecond=0, SecondLevelRetriesConfig Enabled TimeIncrease="00:00:10" NumberOfRetries=3

Also to notice is that each time the failing events are 'successfully' completed, as can be seen on the last-log-output-line. Not sure if these are duplicate deferred messages? also the messages are completed within few seconds (less then lock-time)

Also a note is that when this starts occurring and one tries to stop the NTService, after the last log-line "Shutdown complete." the Service hangs, and the process needs to be "terminated" manually, 
where sometimes there is one log-line coming in after the shutdown-line, i.e. from the EventHandler or ASB error.

Have tried to solve this issue by deleting all topics and the related queues, which solves the issue temporary, where after few hours the issue has returned.

Following is the versions being used,
    NServiceBus Version=5.2.21
    NServiceBus.Autofac Version=5.0.0
    NServiceBus.Azure Version=6.2.3
    NServiceBus.Azure.Transports.WindowsAzureServiceBus Version=6.4.2
    NServiceBus.Host Version=6.0.0
    NServiceBus.Hosting.Azure Version=6.2.4
    NServiceBus.NLog Version=1.1.0

Here follows some of the ASB errors that has been occurring the past day, 

`"Failed to process message System.Transactions.TransactionException: The operation is not valid for the state of the transaction"`
- Could occur time after time within a second a part.

`"A message lock lost exception occured while trying to complete a message, you may consider to increase the lock duration or reduce the batch size, the exception was The lock supplied is invalid. Either the lock expired, or the message has already been removed from the queue. TrackingId:54cf53f4-dd2c-47c2-9ad6-062ce5960d72_G30_B18, SystemTracker:xxx:Topic:xxxx.events|xxxx.xxx, Timestamp:4/3/2018 1:40:29 PM `

**Comment by danielmarbach** [comment]
@danielHalan thanks for reporting this. 

It looks like you previously had a similar issue that you reported in October 2017 (Case Number  00031765). Could they be related?

There was a Service Bus outage on the 27th in WestUs, see https://discuss.particular.net/t/azure-service-bus-outage/465/2 In which region are you running the affected service?

Did a clock drift occur on the client that might have affected the lock renewal?

**Comment by SeanFeldman** [comment]
@danielHalan what @danielmarbach said plus
- What was the outcome of the issue you had previously?
- To share more info, would recommend to open a support case.
- Have you asked Azure support to look into this incident?

**Comment by danielHalan** [comment]
Yes actually related to Case 00031765.

Its in WestEU, and it started to occur on 2-3rd April.

Its a AzureVM, and first thing was to ensure its clock has been synced with Windows servers, so should not been the issue.

Also noticed similar issue today on an different EventHandler, where it got stuck on 2 failing events, which managed to resolve by starting additional VM/EH which started to process the other waiting events, while shutting down the failing one.

**Comment by danielHalan** [comment]
@SeanFeldman 
- Noticed that the issue seemed unresolved, (think I missed your last replies)
- Yes, could be a good idea
- Yes, have opened a ticket at MS/ASB

**Comment by SeanFeldman** [comment]
@danielHalan I'm going to close this issue. Please re-open or raise a new one should you have more details to follow up. Thank you.
